lib/BinaryNode.cpp

```diff
@@ -39,22 +39,20 @@ BinaryNode::BinaryNode(void* p, qore_size_t size) : SimpleValueQoreNode(NT_BINAR
 }
 
 BinaryNode::~BinaryNode() {
-    if (ptr)
+    if (ptr) {
         free(ptr);
+    }
 }
 
 void BinaryNode::clear() {
-    if (len) {
-        assert(ptr);
+    // NOTE: must check 'ptr', len may be 0 with memory allocated
+    if (ptr) {
         free(ptr);
-        len = 0;
+        if (len) {
+            len = 0;
+        }
         ptr = nullptr;
     }
-#ifdef DEBUG
-    else {
-        assert(!ptr);
-    }
-#endif
 }
 
 // returns 0 = equal, 1 = not equal
```

Comment:
it actually returns <0 >0 not only 0/1 http://www.cplusplus.com/reference/cstring/memcmp/